bugsnag-android-core/src/test/java/com/bugsnag/android/MetadataConcurrentModificationTest.kt

```diff
@@ -18,4 +20,30 @@ internal class MetadataConcurrentModificationTest {
             }
         }
     }
+
+    /**
+     * Regression unit test that verifies setting redactedKeys
+     * concurrently on [Metadata] does not throw a ConcurrentModificationException
+     */
+    @Test()
+    fun testConcurrentModificationRedactedKeys() {
+        val keys = mutableSetOf("alpha", "omega")
+        val orig = Metadata()
+        val executor = Executors.newSingleThreadExecutor()
+
+        repeat(100) {
+            orig.redactedKeys = keys
+            executor.execute {
+                keys.add("$it")
+            }
+        }
+    }
+
+    @Test()
+    fun testRedactedKeysCopy() {
+        val orig = Metadata()
+        orig.redactedKeys = mutableSetOf("alpha", "omega")
+        val copy = orig.copy()
+        assertNotSame(orig.redactedKeys, copy.redactedKeys)
```

Comment:
Would be good to have a positive assertion of the copy too: checking for "alpha" and "omega"?